Userland/Libraries/LibWeb/HTML/FrameHostElement.cpp

```diff
@@ -46,8 +48,17 @@ void FrameHostElement::inserted()
     HTMLElement::inserted();
     if (!is_connected())
         return;
-    if (auto* frame = document().frame())
+    if (auto* frame = document().frame()) {
+        auto recursion_level = frame->get_recursion_level() + 1;
+
+        // This seems like a total hack, but it looks as if firefox is doing the same:
+        // Once a certain recursion-threshold is reached it just renders an empty document.
+        if (recursion_level > MAX_RECURSION_DEPTH)
+            document().remove_all_children();
```

Comment:
This is still a total hack though; I think it would be cleaner to simply refuse to load anything into the subframe at the max level, instead of nuking the entire DOM which may have already run arbitrary JavaScript, etc.

Reply:
I put the responsibility into the FrameLoader, which can decide to not even load the content if the recursion threshold is exceeded for the frame.